web/packages/teleport/src/Discover/ConnectMyComputer/TestConnection/TestConnection.tsx

```diff
@@ -0,0 +1,106 @@
+/**
+ * Copyright 2023 Gravitational, Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React, { useEffect, useCallback, useRef } from 'react';
+
+import { ButtonPrimary, Flex, Indicator, Text } from 'design';
+import * as Icons from 'design/Icon';
+import { useAsync } from 'shared/hooks/useAsync';
+
+import useTeleport from 'teleport/useTeleport';
+import {
+  ActionButtons,
+  StyledBox,
+  Header,
+  TextIcon,
+} from 'teleport/Discover/Shared';
+import { NodeConnect } from 'teleport/UnifiedResources/ResourceActionButton';
+
+import { NodeMeta } from '../../useDiscover';
+
+import type { AgentStepProps } from '../../types';
+
+export const TestConnection = (props: AgentStepProps) => {
+  const { userService } = useTeleport();
+  const abortController = useRef<AbortController>();
+  const [reloadUserAttempt, reloadUser] = useAsync(
```

Comment:
I used `useAsync` instead of Suspense because I realized that Suspense is currently undocumented and basically depends on you using some kind of an abstraction. We seem to be using Suspense through that undocumented API and I couldn't find any abstraction on top of that. As I'm tight on time, I opted to use what I know to achieve the same result.

Let me know if I should rewrite it to use Suspense. I can do it once I get around to the connection test. But tbh, I don't see the value of doing that without any abstractions – it's significantly more code which, to my eye, also seems more complex and I don't see what benefits we get in return.

Reply:
i am curious where you got that you had to use `suspense`? the only suspense we use is with [useJoinTokenSuspender](https://github.com/gravitational/teleport/blob/28f57ae2093f53395ca41593b8d07e328a7c7dbe/web/packages/teleport/src/Discover/Shared/useJoinTokenSuspender.ts) as far as i know (it's for join scripts). tbh, it was pretty painful to use suspense because it was a completely different way of thinking/structuring your code

Reply:
Yep, that was the hook I saw. I saw it first in `DownloadScript` so I assumed that this is the way to do things if we want something to happen before a step gets rendered.

I think using Suspense there would be easier if:

1) It was hidden behind some kind of abstraction so that the hook itself doesn't have to implement all this fancy stuff with throwing promises etc.
2) The hook usage was scoped as tightly as possible to the component that's actually going to use this data so that we don't have to go out of our way with the `Template` component.

But those are just my observations based on the existing code, I've never actually used Suspense so I don't know what the best practices are really.